context-free.md

```diff
@@ -2,21 +2,28 @@
 
 ##  Who is this programming language for?
 
+I think this language is for people that want to make digital art using context-free grammars.
 
 ## What is easy to do in this language? Why is it easy?
 
+It is easy to create rules in this language. It is also easy to use primitive shapes and modify them. These kinds of things are easy because they are the bread and butter of the language. Everything needs rules and primitive shapes along with the parameters for the shapes.
 
 ## What is hard to do in this language? Why is it hard?
 
+I think it is hard to keep track of all the rules and where everything is going to show up on the screen. Although beautiful, complex diagrams can be created recursively, there are probably times where one will need a lot of rules to make the drawings they want. Rules can quickly become incoherent if the number of rules and dependencies grows large. The coordinate system and sizing of the objects is a bit hard as well. It's hard to think of coordinates and know where they are going to exactly be placed. It would sometimes take several tries to get a shape in the exact desired place. The sizing of the objects is also a bit hard to intuit because the program can resize the drawing to fit the screen. There could be something to stop this from happening, but this is just from my initial experience.
```

Comment:
I had a similar feeling.  I wonder if it's designed for making images that require only a small number of lines of code to make or if there is some framework that makes it easier to keep track of everything